nnvm/python/nnvm/testing/check_computation.py

```diff
@@ -0,0 +1,451 @@
+# pylint: disable=cell-var-from-loop
+"""Helper utilities to check functions and their gradients."""
+from __future__ import absolute_import as _abs
+
+import logging
+import numpy as np
+
+import tvm
+from tvm.contrib import graph_runtime
+
+import nnvm
+from nnvm.compiler import graph_util
+from nnvm.compiler.graph_attr import TCODE_TO_DTYPE
+from .config import ctx_list
+
+def graph_to_function(graph, target, ctx):
+    """Convert a graph to a function taking a keyword args and returning a list of results
+    (both args and results are numpy arrays).
+
+    Example::
+
+        fun = graph_to_function(graph, llvm, cpu(0))
+        [res1, res2] = fun(x=np.zeros((1,2)), y=np.zeros((1,)))
+
+    Parameters
+    ----------
+    graph : nnvm.graph.Graph
+        A graph we want to convert to a function.
+
+    target : str or :any:`tvm.target.Target`, optional
+        The build target
+
+    ctx : TVMContext
+        The context to deploy the module.
+
+    Returns
+    -------
+    function : Callable[..., List[numpy.ndarray]]
+    """
+
+    shapes = graph.json_attr('shape')
+    dtypes = graph.json_attr('dtype')
+
+    if shapes is None or dtypes is None:
+        graph = graph.apply('InferShape').apply('InferType')
```

Comment:
Here we assume set_shape_inputs and set_dtype_inputs have been called. It might be better to directly pass input_shapes and input_types into this function to make it more general.